fava/core/file.py

```diff
@@ -110,6 +110,24 @@ def insert_entries(self, entries):
                          self.ledger.fava_options['insert-entry'])
             self.ledger.extensions.run_hook('after_insert_entry', entry)
 
+    def render_entries(self, entries):
+        """Return entries in Beancount format.
+
+        Only renders entries that are in the original Beancount file(s).
+
+        Args:
+            entries: A list of entries.
+
+        Returns:
+            A string representing the entries rendered in Beancount format.
+
+        """
+        entries = [get_entry_slice(entry)[0] for entry in entries
```

Comment:
This might fail if the entry is plugin-generated. We could check two lines below if `entry.meta.get('filename')` is one of the files in the ledger.

Or maybe we would want to include plugin-generated entries too? Depending on the setup simply ignoring those entries on export might lead to a useless export.

Reply:
> Or maybe we would want to include plugin-generated entries too?

I think we want - but I have no idea how to do that, as right now `get_entry_slice` looks for the "source" in the beancount file. In the case of plugin-generated entries we would have to render an entry from the Python object.

Reply:
Sorry, somehow missed this comment. You can use `_format_entry` for entries where get_entry_slice fails. Find out with which exceptions it can fail and wrap it in a try-except block. (Maybe that's easier to do if this function for-loops over the entries `yield`s entry slices and the template joins them)

Reply:
I implemented this in a rudimentary way, ignoring "Summarize"-entries (because I don't want to have them).